### ISSUE TITLE
Faceted Browser UTF-8 encoding issue

### ISSUE THREAD

**Issue by jakubklimek** [openlink/virtuoso-opensource#141]
Property and Object labels have a wrong encoding of international characters.
fct 1.13.56 in 7d38d124560a8df1802edd1dccc534b76c21d044

See the second object of the last property (should be Březiněves, as it is when I click on it - the second screenshot)
![image](https://f.cloud.github.com/assets/718827/2243767/396f7b30-9d33-11e3-85a1-db0753631d33.png)
http://ruian.linked.opendata.cz/resource/obce/554782

![image](https://f.cloud.github.com/assets/718827/2243771/4e427274-9d33-11e3-8dc9-35768d89584e.png)
http://ruian.linked.opendata.cz/resource/katastralni-uzemi/614131

However, on the last page of http://ruian.linked.opendata.cz/resource/katastralni-uzemi/614131 (http://ruian.linked.opendata.cz/describe/?url=http%3A%2F%2Fruian.linked.opendata.cz%2Fresource%2Fkatastralni-uzemi%2F614131&p=1&lp=15&op=-1&last=&gp=1) the same characters are OK, which seems a bit strange:
![image](https://f.cloud.github.com/assets/718827/2243910/a69b0902-9d35-11e3-8c32-a6071286c3a2.png)

**Comment by HughWilliams** [comment]
Is this still an issue as I do not see the utf-8 encoding issues on your screenshots when I access the live pages on my browser ?

/Users/hwilliams/Dropbox/Screenshots/Screenshot 2014-03-09 13.29.35.png

**Comment by jakubklimek** [comment]
I still have this issue when I access http://ruian.linked.opendata.cz/resource/obce/554782
![image](https://f.cloud.github.com/assets/718827/2371880/6091e9ca-a830-11e3-9dc9-45899c18124a.png)
in fct 1.13.57

**Comment by jakubklimek** [comment]
This is still happening in 8173801092e9fd4abc06d32e642e6d7d72b4bf40 in fct 1.13.59 http://ruian.linked.opendata.cz/resource/obce/554782 
![image](https://cloud.githubusercontent.com/assets/718827/3284856/8ca69e22-f531-11e3-91a3-161a6a07e84e.png)

**Comment by HughWilliams** [comment]
Has anything changed in the Virtuoso Server instance as the page http://ruian.linked.opendata.cz/resource/obce/554782  , does not load with the corrupt characters as in your screenshot above:

![screenshot 2014-06-22 21 37 44](https://cloud.githubusercontent.com/assets/4868081/3352496/bec66770-fa4a-11e3-8c3e-35aefa362fad.png)

**Comment by jakubklimek** [comment]
No, but the texts are in Czech. So when I view it from here, they load because of my location I guess and they don't load for you. I think that adding &lang=cs to the url should do the trick.

**Comment by jakubklimek** [comment]
@HughWilliams This is still happening, database <a href="http://ruian.linked.opendata.cz/db/dbruian.zip">here</a>, dont forget the &lang=cs as you will not be autodetected to be in cs environment.

**Comment by jakubklimek** [comment]
Plus, now it looks like this: (notice the font):
![image](https://cloud.githubusercontent.com/assets/718827/5788225/255f8976-9e29-11e4-9880-294672643b5a.png) which is probably the same bug as was fixed in a614876e4fc02a1f9ee256f5492dd357f8c648fa however that was in sparql endpoint, in fct it is still there. However, only in some Chrome instances, which is weird.

**Comment by HughWilliams** [comment]
@jakubklimek: What is the URL to load as assume it http://localhost:8890/resource/obce/554782  having setup the test database on http://localhost:8890 , but I get a 404 error:

Error HTTP/1.1 404 File not found
The requested URL was not found    URI  = '/resource/obce/554782'

**Comment by jakubklimek** [comment]
@HughWilliams That is rewritten by Apache to form correct and resolvable Linked Data URIs. For your instance it will be <a href="http://localhost:8890/describe/?url=http://ruian.linked.opendata.cz/resource/obce/554782&lang=cs">http://localhost:8890/describe/?url=http://ruian.linked.opendata.cz/resource/obce/554782&lang=cs</a>

**Comment by HughWilliams** [comment]
Loading the page  http://localhost:8890/describe/?url=http://ruian.linked.opendata.cz/resource/obce/554782&lang=cs , I do not see the UTF-8 encoding issue you report, see screen shot attached:

![screen shot 2015-01-24 at 15 16 44](https://cloud.githubusercontent.com/assets/4868081/5887802/4b02dabc-a3dc-11e4-80d3-e714c97f7942.png)

**Comment by jakubklimek** [comment]
What about the fourth property... there si an errorneous character on the
line containing MOMC

**Comment by HughWilliams** [comment]
@jakubklimek: This looks like a browser rendering problem as if I view the source of the page I see:

title="Typ MOMC, na něž je statutární město rozčleněno">Typ MOMC, na něž...�sto rozčleněno</a>

**Comment by jakubklimek** [comment]
So the weird character is generated by browser?

Anyway, I have narrowed it down to an issue in combination with apache mod_proxy_html... I am using that to do the URI resolvablility. When I display the page directly via /describe/?url=... It looks like on your screenshot. However, when I use the resolvable uri via Apache, it is transcoded for no apparent reason - the output document is still utf-8 encoded, but all the special czech characters are destroyed. I am playing with Apache mod_proxy_html now but could not find a solution yet

**Comment by jakubklimek** [comment]
OK, I found the solution and tested it with a copy of the html page generated by fct. The issue is described here: http://serverfault.com/questions/559518/mod-proxy-html-garbles-non-ascii-characters

Could you add the missing <code>&lt;meta http-equiv="Content-Type" content="text/html; charset=UTF-8"&gt;</code> tag to fct generated XHTML output? When this is added there, mod_proxy_html detects the encoding correctly and does not destroy the characters.

And the same problem goes for HTML generated by the SPARQL endpoint.

**Comment by jakubklimek** [comment]
@HughWilliams Any possibility to fix this by adding the line mentioned in my previous comment? It is quite simple fix. I have tried many times to fix this using various workarounds in Apache but failed.
